daemon/src/daemon.rs

```diff
@@ -569,26 +574,32 @@ impl DocumentActor {
                         let relative_file_path =
                             RelativePath::try_from_path(&self.base_dir, file_path)
                                 .expect("Walked file path should be within base directory");
-                        if let Ok(text) = String::from_utf8(bytes) {
-                            if init {
-                                self.crdt_doc.initialize_text(&text, &relative_file_path);
+                        if self.owns(&relative_file_path) {
```

Comment:
Note to any other reader: The diff is confusing, here we just insert this line around the unchanged `if let Ok(text) ...` block.